operator-common/src/test/java/io/strimzi/operator/PlatformFeaturesAvailabilityTest.java

```diff
@@ -7,157 +7,155 @@
-                "  \"minor\": \"20\",\n" +
-                "  \"gitVersion\": \"v1.20.1\",\n" +
-                "  \"gitCommit\": \"c4d752765b3bbac2237bf87cf0b1c2e307844666\",\n" +
-                "  \"gitTreeState\": \"clean\",\n" +
-                "  \"buildDate\": \"2020-12-18T12:09:25Z\",\n" +
-                "  \"goVersion\": \"go1.15.5\",\n" +
-                "  \"compiler\": \"gc\",\n" +
-                "  \"platform\": \"linux/amd64\"\n" +
-                "}";
-
-        HttpServer mockHttp = startMockApi(context, version, Collections.EMPTY_LIST);
-
-        KubernetesClient client = new DefaultKubernetesClient("127.0.0.1:" + mockHttp.actualPort());
-
-        Checkpoint async = context.checkpoint();
-
-        PlatformFeaturesAvailability.create(vertx, client).onComplete(context.succeeding(pfa -> context.verify(() -> {
-            assertThat("Versions are not equal", pfa.getKubernetesVersion(), is(KubernetesVersion.V1_20));
-            stopMockApi(context, mockHttp);
-            async.flag();
-        })));
+    public void testVersionDetectionOpenShift(VertxTestContext context) {
+        version = () ->  "{\n" +
+                        "  \"major\": \"1\",\n" +
+                        "  \"minor\": \"20\",\n" +
+                        "  \"gitVersion\": \"v1.20.1\",\n" +
+                        "  \"gitCommit\": \"c4d752765b3bbac2237bf87cf0b1c2e307844666\",\n" +
+                        "  \"gitTreeState\": \"clean\",\n" +
+                        "  \"buildDate\": \"2020-12-18T12:00:47Z\",\n" +
+                        "  \"goVersion\": \"go1.15.5\",\n" +
+                        "  \"compiler\": \"gc\",\n" +
+                        "  \"platform\": \"linux/amd64\"\n" +
+                        "}";
+
+        verify(context, pfa -> assertThat("Versions are not equal", pfa.getKubernetesVersion(), is(KubernetesVersion.V1_20)));
     }
 
     @Test
-    public void testApiDetectionOce(VertxTestContext context) throws InterruptedException {
-        List<String> apis = new ArrayList<>();
-        apis.add("/apis/route.openshift.io/v1");
-        apis.add("/apis/build.openshift.io/v1");
+    public void testVersionDetectionMinikube(VertxTestContext context) {
+        version = () -> "{\n" +
+                        "  \"major\": \"1\",\n" +
+                        "  \"minor\": \"20\",\n" +
+                        "  \"gitVersion\": \"v1.20.1\",\n" +
+                        "  \"gitCommit\": \"c4d752765b3bbac2237bf87cf0b1c2e307844666\",\n" +
+                        "  \"gitTreeState\": \"clean\",\n" +
+                        "  \"buildDate\": \"2020-12-18T12:09:25Z\",\n" +
+                        "  \"goVersion\": \"go1.15.5\",\n" +
+                        "  \"compiler\": \"gc\",\n" +
+                        "  \"platform\": \"linux/amd64\"\n" +
+                        "}";
+
+        verify(context, pfa ->  assertThat("Versions are not equal", pfa.getKubernetesVersion(), is(KubernetesVersion.V1_20)));
```

Comment:
What is the point of these tests? 
* Why do they check which version it isn't instead of which version it is?
* How do these two tests differ? The version info seems to be exactly the same

Reply:
I'm not sure, I wasn't evaluating the test cases, just refactoring the plumbing. If it's not going to obscure my desired changes, happy to look into them in PR V2 Electric Boogaloo.